src/main/java/org/students/simplebitcoinnode/controller/TransactionsController.java

```diff
@@ -38,21 +39,20 @@ public TransactionsController(TransactionService transactionService) {
      * @throws IllegalArgumentException if the provided type is null or empty.
      */
     @GetMapping("/transactions")
-    public List<Transaction> getTransactions(@RequestParam String pubKey, @RequestParam String type) {
-        return transactionService.getTransactions(pubKey, type);
+    public ResponseEntity<List<GetTransactionDTO>> getTransactions(@RequestParam String pubKey, @RequestParam String type) {
+        return ResponseEntity.ok().body(transactionService.getTransactions(pubKey, type));
     }
 
     /**
      * This method is used to create new transactions.
      *
-     * @param transaction This is the transaction to be created.
+     * @param newTransactionDTO This is the transaction to be created.
      * @return returns the result of the transaction creation.
      */
+    @Valid
     @PostMapping("/send")
-    public ResponseEntity<?> newTransactions(@Valid @RequestBody Transaction transaction) {
-        // Custom constraint validation using @Valid annotation on the arguments works as expected.
-        // All custom ConstraintValidators have their dependencies injected as expected.
-        return ResponseEntity.ok().body(transactionService.newTransactions(transaction));
+    public ResponseEntity<Integer> newTransactions(@RequestParam NewTransactionDTO newTransactionDTO){
```

Comment:
Already there. 
![image](https://github.com/TheGreyCore/SimpleBitcoinNode/assets/104592742/5328838b-85e7-4f44-9340-f5cce2d3732b)

Reply:
That's not the correct usage of `@Valid` annotation because it does not validate the request parameter but instead validates the controller return object. Apply `@Valid` annotation to the method argument instead. Also I just noticed that you are using `@RequestParam` for controller argument, which essentially maps url variables to the object, which is not the desired usage of the REST API. Data about new transactions should be contained in the request body and as such please replace this annotation with `@RequestBody`.

TL;DR do this:
```java
public ResponseEntity<?> newTransactions(@Valid @RequestBody NewTransactionDTO newTransactionDTO) {
```

Reply:
Mhm, fixed typo